tests/Support/FakeAdapter.php

```diff
@@ -40,8 +41,13 @@ public function subscribe(callable $handlerCallback): void
         // TODO: Implement subscribe() method.
     }
 
-    public function withChannel(string $channel): AdapterInterface
+    public function withChannel(BackedEnum|string $channel): AdapterInterface
     {
         // TODO: Implement withChannel() method.
     }
+
+    public function getChannel(): string
+    {
+        // TODO: Implement getChannel() method.
```

Comment:
Should we do a dummy string return here?

Reply:
Or throw `LogicException`